addons/account/wizard/account_resequence.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools.date_utils import get_month, get_fiscal_year
+from odoo.tools.misc import format_date
+
+import re
+from collections import defaultdict
+import json
+
+
+class ReSequenceWizard(models.TransientModel):
+    _name = 'account.resequence.wizard'
+    _description = 'Remake the sequence of Journal Entries.'
+
+    sequence_number_reset = fields.Char(compute='_compute_sequence_number_reset')
+    first_date = fields.Date(help="Date (inclusive) from which the numbers are resequenced.")
+    end_date = fields.Date(help="Date (inclusive) to which the numbers are resequenced. If not set, all Journal Entries up to the end of the period are resequenced.")
+    first_name = fields.Char(compute="_compute_first_name", readonly=False, store=True, required=True, string="First New Sequence")
+    ordering = fields.Selection([('keep', 'Keep current order'), ('date', 'Reorder by accounting date')], required=True, default='keep')
+    move_ids = fields.Many2many('account.move')
+    new_values = fields.Text(compute='_compute_new_values')
+    preview_moves = fields.Text(compute='_compute_preview_moves')
+
+    @api.model
+    def default_get(self, fields_list):
+        values = super(ReSequenceWizard, self).default_get(fields_list)
+        active_move_ids = self.env['account.move']
+        if self.env.context.get('select_all') and self.env.context.get('active_model') == 'account.move' and 'active_domain' in self.env.context:
+            active_move_ids = self.env['account.move'].search(self.env.context['active_domain'])
+        elif self.env.context['active_model'] == 'account.move' and 'active_ids' in self.env.context:
+            active_move_ids = self.env['account.move'].browse(self.env.context['active_ids'])
+        if len(active_move_ids.journal_id) > 1:
+            raise UserError(_('You can only resequence items from the same journal'))
```

Comment:
raise if credit notes and invoices of the same journal are selected but the journal says they should have a different numbering